src/boot/efi/boot.c

```diff
@@ -1871,6 +1872,101 @@ static bool is_sd_boot(EFI_FILE *root_dir, const char16_t *loader_path) {
         return memcmp(content, magic, sizeof(magic)) == 0;
 }
 
+#if defined(__i386__) || defined(__x86_64__)
+#  define EDK2_FIRMWARE_VOLUME_GUID \
+        GUID_DEF(0x7cb8bdc9, 0xf8eb, 0x4f34, 0xaa, 0xea, 0x3e, 0xe4, 0xaf, 0x65, 0x16, 0xa1)
+#elif defined(__arm__) || defined(__aarch64__)
+#  define EDK2_FIRMWARE_VOLUME_GUID \
+        GUID_DEF(0x64074afe, 0x340a, 0x4be6, 0x94, 0xba, 0x91, 0xb5, 0xb4, 0xd0, 0xf7, 0x1e)
+#elif defined(__riscv)
+#  define EDK2_FIRMWARE_VOLUME_GUID \
+        GUID_DEF(0x27a72e80, 0x3118, 0x4c0c, 0x86, 0x73, 0xaa, 0x5b, 0x4e, 0xfa, 0x96, 0x13)
+#endif
+
+#ifdef EDK2_FIRMWARE_VOLUME_GUID
+static struct {
+        VENDOR_DEVICE_PATH fv;
+        VENDOR_DEVICE_PATH ff;
+        EFI_DEVICE_PATH end;
+} _packed_ edk2_firmware_uefi_shell_dp = {
+        .fv = {
+                .Header = {
+                        .Type = MEDIA_FILEPATH_DP,
+                        .SubType = MEDIA_PIWG_FW_VOL_DP,
+                        .Length = sizeof(edk2_firmware_uefi_shell_dp.fv),
+                },
+                .Guid = EDK2_FIRMWARE_VOLUME_GUID,
+        },
+        .ff = {
+                .Header = {
+                        .Type = MEDIA_FILEPATH_DP,
+                        .SubType = MEDIA_PIWG_FW_FILE_DP,
+                        .Length = sizeof(edk2_firmware_uefi_shell_dp.ff),
+                },
+                .Guid = UEFI_SHELL_GUID,
+        },
+        .end = DEVICE_PATH_END_NODE,
+};
```

Comment:
please move the guid and struct to some new header, out of boot.c. it's already wayy too long